cmd/kube-scheduler/app/options/options.go

```diff
@@ -238,9 +238,13 @@ func (o *Options) Config() (*schedulerappconfig.Config, error) {
 
 	// Prepare event clients.
 	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{Interface: eventClient.EventsV1beta1().Events("")})
-	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, c.ComponentConfig.SchedulerName)
+	reportingController := ""
+	if c.ComponentConfig.SchedulerName != nil {
```

Comment:
instead of defaulting to "", we should error fatal if schedulerName is not set or empty string.